huaweicloud/services/apig/resource_huaweicloud_apig_endpoint_connection_management.go

```diff
@@ -0,0 +1,235 @@
+	return &schema.Resource{
+		CreateContext: resourceEndpointConnectionManagementCreate,
+		ReadContext:   resourceEndpointConnectionManagementRead,
+		UpdateContext: resourceEndpointConnectionManagementUpdate,
+		DeleteContext: resourceEndpointConnectionManagementDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"instance_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Specifies the ID of the dedicated instance to which the endpoint connection belongs.",
+			},
+			"endpoint_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Specifies the ID of the endpoint connection.",
+			},
+			"action": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Specifies the operation type endpoint connection.",
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The current ststus of the endpoint connection.",
+			},
+		},
+	}
+}
+
+func resourceEndpointConnectionManagementCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	client, err := cfg.NewServiceClient("apig", cfg.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating APIG Client: %s", err)
+	}
+
+	err = approveEndpointConnection(ctx, client, d, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(d.Get("endpoint_id").(string))
+	return resourceEndpointConnectionManagementRead(ctx, d, meta)
+}
+
+func approveEndpointConnection(ctx context.Context, client *golangsdk.ServiceClient, d *schema.ResourceData, timeout time.Duration) error {
+	var (
+		httpUrl    = "v2/{project_id}/apigw/instances/{instance_id}/vpc-endpoint/connections/action"
+		instanceId = d.Get("instance_id").(string)
+		endpointId = d.Get("endpoint_id").(string)
+		action     = d.Get("action").(string)
+	)
+	path := client.Endpoint + httpUrl
+	path = strings.ReplaceAll(path, "{project_id}", client.ProjectID)
+	path = strings.ReplaceAll(path, "{instance_id}", instanceId)
+	opts := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody: map[string]interface{}{
+			"action":    action,
+			"endpoints": []string{endpointId},
+		},
+	}
+	_, err := client.Request("POST", path, &opts)
+	if err != nil {
+		actionStatus := map[string]string{
+			"receive": "receiving",
+			"reject":  "rejecting",
+		}
+		return fmt.Errorf("error %s endpoint connection (%s) under dedicated instance (%s): %s", actionStatus[action], endpointId, instanceId, err)
```

Comment:
```suggestion
		return fmt.Errorf("failed to %s endpoint connection (%s) under dedicated instance (%s): %s", actionStatus[action], endpointId, instanceId, err)
```
please using `failed to do sth` to instead, if you want to use the `error` to describe it, that it is the `error doing sth`.

Reply:
fixed
![image](https://github.com/huaweicloud/terraform-provider-huaweicloud/assets/150208787/241910a5-ce0d-412b-9477-33679dd41584)